src/MetaCompilation/MetaCompilation/MetaCompilation/DiagnosticAnalyzer.cs

```diff
@@ -1302,317 +1294,315 @@ private void IfDiagnostic(CompilationAnalysisContext context, StatementSyntax st
+                            List<string> spanNames = SpanAnalysis(statements[statementCount - 3], locationCorrect);
+                            if (spanNames.Count == 0 || spanNames[0] == "" || spanNames[1] == "")
                             {
-                                IfDiagnostic(context, statements[6], SpanIncorrectRule, startToken.Text, endToken.Text);
+                                IfDiagnostic(context, statements[statementCount - 3], SpanIncorrectRule, locationCorrect);
                                 return false;
                             }
 
-                            if (statementCount > 7)
+                            if (statementCount > 6)
                             {
-                                SyntaxToken locationToken = LocationAnalysis(statementIdentifierToken, spanToken, statements);
-                                if (locationToken.Text == "")
+                                bool startSpanCorrect = StartAnalysis(keywordIdentifierToken, statements, spanNames[0]);
+                                bool endSpanCorrect = EndAnalysis(statementIdentifierToken, statements, spanNames[1]);
+                                if (!endSpanCorrect && startSpanCorrect)
+                                {
+                                    if (statementCount == 7)
+                                    {
+                                        var argument = ((statements[4] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[1];
+                                        ReportDiagnostic(context, EndSpanMissingRule, argument.GetLocation(), locationCorrect, statementIdentifierToken.Text);
+                                        return false;
+                                    }
+                                    else
+                                    {
+                                        var argument = ((statements[5] as LocalDeclarationStatementSyntax).Declaration.Variables[0].Initializer.Value as InvocationExpressionSyntax).ArgumentList.Arguments[1];
+                                        ReportDiagnostic(context, EndSpanMissingRule, argument.GetLocation(), locationCorrect, statementIdentifierToken.Text);
+                                        return false;
+                                    }
+                                }
+                                else if (!endSpanCorrect && !startSpanCorrect)
                                 {
-                                    IfDiagnostic(context, statements[7], LocationIncorrectRule, statementIdentifierToken.Text, spanToken.Text);
+                                    IfDiagnostic(context, statements[3], EndSpanIncorrectRule, statementIdentifierToken.Text);
                                     return false;
                                 }
 
-                                if (statementCount > 8)
+                                if (!startSpanCorrect)
```

Comment:
@jepetty this should really be an else if on the if statement above it, and should explicitly say the case that it is covering, so else if(endSpanCorrect && !startSpanCorrect)